# Frameworks


In [145]:
import pandas as pd
import plotly.express as px

# Load Dataset

In [146]:
df=pd.read_csv("amazon.csv")
df.head()

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count,about_product,user_id,user_name,review_id,review_title,review_content,img_link,product_link
0,B07JW9H4J1,Wayona Nylon Braided USB to Lightning Fast Cha...,Computers&Accessories|Accessories&Peripherals|...,₹399,"₹1,099",64%,4.2,"24,269",High Compatibility : Compatible With iPhone 12...,"AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...","Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...","R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...","Satisfied,Charging is really fast,Value for mo...",Looks durable Charging is fine tooNo complains...,https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...
1,B098NS6PVG,Ambrane Unbreakable 60W / 3A Fast Charging 1.5...,Computers&Accessories|Accessories&Peripherals|...,₹199,₹349,43%,4.0,"43,994","Compatible with all Type C enabled devices, be...","AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...","ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...","RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...","A Good Braided Cable for Your Type C Device,Go...",I ordered this cable to connect my phone to An...,https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/Ambrane-Unbreakable-Char...
2,B096MSW6CT,Sounce Fast Phone Charging Cable & Data Sync U...,Computers&Accessories|Accessories&Peripherals|...,₹199,"₹1,899",90%,3.9,"7,928",【 Fast Charger& Data Sync】-With built-in safet...,"AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...","Kunal,Himanshu,viswanath,sai niharka,saqib mal...","R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...","Good speed for earlier versions,Good Product,W...","Not quite durable and sturdy,https://m.media-a...",https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/Sounce-iPhone-Charging-C...
3,B08HDJ86NZ,boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...,Computers&Accessories|Accessories&Peripherals|...,₹329,₹699,53%,4.2,"94,363",The boAt Deuce USB 300 2 in 1 cable is compati...,"AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...","Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...","R3EEUZKKK9J36I,R3HJVYCLYOY554,REDECAZ7AMPQC,R1...","Good product,Good one,Nice,Really nice product...","Good product,long wire,Charges good,Nice,I bou...",https://m.media-amazon.com/images/I/41V5FtEWPk...,https://www.amazon.in/Deuce-300-Resistant-Tang...
4,B08CF3B7N1,Portronics Konnect L 1.2M Fast Charging 3A 8 P...,Computers&Accessories|Accessories&Peripherals|...,₹154,₹399,61%,4.2,"16,905",[CHARGE & SYNC FUNCTION]- This cable comes wit...,"AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...","rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...","R1BP4L2HH9TFUP,R16PVJEXKV6QZS,R2UPDB81N66T4P,R...","As good as original,Decent,Good one for second...","Bought this instead of original apple, does th...",https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/Portronics-Konnect-POR-1...


# Dataset Overview


In [147]:
print("Data Shape",df.shape)
print("Size of data",df.size)

Data Shape (1465, 16)
Size of data 23440


In [148]:
print(df.columns)

Index(['product_id', 'product_name', 'category', 'discounted_price',
       'actual_price', 'discount_percentage', 'rating', 'rating_count',
       'about_product', 'user_id', 'user_name', 'review_id', 'review_title',
       'review_content', 'img_link', 'product_link'],
      dtype='object')


In [149]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   product_id           1465 non-null   object
 1   product_name         1465 non-null   object
 2   category             1465 non-null   object
 3   discounted_price     1465 non-null   object
 4   actual_price         1465 non-null   object
 5   discount_percentage  1465 non-null   object
 6   rating               1465 non-null   object
 7   rating_count         1463 non-null   object
 8   about_product        1465 non-null   object
 9   user_id              1465 non-null   object
 10  user_name            1465 non-null   object
 11  review_id            1465 non-null   object
 12  review_title         1465 non-null   object
 13  review_content       1465 non-null   object
 14  img_link             1465 non-null   object
 15  product_link         1465 non-null   object
dtypes: obj

# Data Cleanning

## Drop Un-Wanted Columns

In [150]:
df.drop(
    columns=[
        "img_link",
        "product_link",
        "review_content",
        "review_title",
        "review_id",
        "user_name",
        "user_id",
        "about_product"
    ],inplace=True
)

In [151]:
df.isnull().sum()

product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 0
rating_count           2
dtype: int64

In [152]:
df[df["rating_count"].isnull()]

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count
282,B0B94JPY2N,Amazon Brand - Solimo 65W Fast Charging Braide...,Computers&Accessories|Accessories&Peripherals|...,₹199,₹999,80%,3.0,NaN
324,B0BQRJ3C47,"REDTECH USB-C to Lightning Cable 3.3FT, [Apple...",Computers&Accessories|Accessories&Peripherals|...,₹249,₹999,75%,5.0,NaN


In [153]:
df = df.dropna()

In [154]:
df.shape

(1463, 8)

In [155]:
df.isnull().sum()

product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 0
rating_count           0
dtype: int64

## Fix data types

##### Datatype change Required

###### 1.discounted_price->float

###### 2.actual_price->float

###### 3.discount_percentage->int

###### 4.rating->float
###### 5.rating_count->int

In [156]:
# Price
price_columns = ["discounted_price", "actual_price"]

df[price_columns] = (
    df[price_columns]
    .replace({"₹": "", ",": ""}, regex=True)
    .astype(float)
)

print(df)

      product_id                                       product_name  \
0     B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1     B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2     B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3     B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4     B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   
...          ...                                                ...   
1460  B08L7J3T31  Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...   
1461  B01M6453MB  Prestige Delight PRWO Electric Rice Cooker (1 ...   
1462  B009P2LIL4  Bajaj Majesty RX10 2000 Watts Heat Convector R...   
1463  B00J5DYCCA  Havells Ventil Air DSP 230mm Exhaust Fan (Pist...   
1464  B01486F4G6  Borosil Jumbo 1000-Watt Grill Sandwich Maker (...   

                                               category  discounted_price  \
0     Computers&Accessories|Accessories&Peripherals|...             39

In [157]:
df

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count
0,B07JW9H4J1,Wayona Nylon Braided USB to Lightning Fast Cha...,Computers&Accessories|Accessories&Peripherals|...,399.0,1099.0,64%,4.2,"24,269"
1,B098NS6PVG,Ambrane Unbreakable 60W / 3A Fast Charging 1.5...,Computers&Accessories|Accessories&Peripherals|...,199.0,349.0,43%,4.0,"43,994"
2,B096MSW6CT,Sounce Fast Phone Charging Cable & Data Sync U...,Computers&Accessories|Accessories&Peripherals|...,199.0,1899.0,90%,3.9,"7,928"
3,B08HDJ86NZ,boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...,Computers&Accessories|Accessories&Peripherals|...,329.0,699.0,53%,4.2,"94,363"
4,B08CF3B7N1,Portronics Konnect L 1.2M Fast Charging 3A 8 P...,Computers&Accessories|Accessories&Peripherals|...,154.0,399.0,61%,4.2,"16,905"
...,...,...,...,...,...,...,...,...
1460,B08L7J3T31,Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...,Home&Kitchen|Kitchen&HomeAppliances|WaterPurif...,379.0,919.0,59%,4,"1,090"
1461,B01M6453MB,Prestige Delight PRWO Electric Rice Cooker (1 ...,Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...,2280.0,3045.0,25%,4.1,"4,118"
1462,B009P2LIL4,Bajaj Majesty RX10 2000 Watts Heat Convector R...,"Home&Kitchen|Heating,Cooling&AirQuality|RoomHe...",2219.0,3080.0,28%,3.6,468
1463,B00J5DYCCA,Havells Ventil Air DSP 230mm Exhaust Fan (Pist...,"Home&Kitchen|Heating,Cooling&AirQuality|Fans|E...",1399.0,1890.0,26%,4,"8,031"


In [158]:
df.discount_percentage = (
    df.discount_percentage
    .replace({"%": "", ",": ""}, regex=True)
    .astype(float)
)

In [159]:
df

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count
0,B07JW9H4J1,Wayona Nylon Braided USB to Lightning Fast Cha...,Computers&Accessories|Accessories&Peripherals|...,399.0,1099.0,64.0,4.2,"24,269"
1,B098NS6PVG,Ambrane Unbreakable 60W / 3A Fast Charging 1.5...,Computers&Accessories|Accessories&Peripherals|...,199.0,349.0,43.0,4.0,"43,994"
2,B096MSW6CT,Sounce Fast Phone Charging Cable & Data Sync U...,Computers&Accessories|Accessories&Peripherals|...,199.0,1899.0,90.0,3.9,"7,928"
3,B08HDJ86NZ,boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...,Computers&Accessories|Accessories&Peripherals|...,329.0,699.0,53.0,4.2,"94,363"
4,B08CF3B7N1,Portronics Konnect L 1.2M Fast Charging 3A 8 P...,Computers&Accessories|Accessories&Peripherals|...,154.0,399.0,61.0,4.2,"16,905"
...,...,...,...,...,...,...,...,...
1460,B08L7J3T31,Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...,Home&Kitchen|Kitchen&HomeAppliances|WaterPurif...,379.0,919.0,59.0,4,"1,090"
1461,B01M6453MB,Prestige Delight PRWO Electric Rice Cooker (1 ...,Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...,2280.0,3045.0,25.0,4.1,"4,118"
1462,B009P2LIL4,Bajaj Majesty RX10 2000 Watts Heat Convector R...,"Home&Kitchen|Heating,Cooling&AirQuality|RoomHe...",2219.0,3080.0,28.0,3.6,468
1463,B00J5DYCCA,Havells Ventil Air DSP 230mm Exhaust Fan (Pist...,"Home&Kitchen|Heating,Cooling&AirQuality|Fans|E...",1399.0,1890.0,26.0,4,"8,031"


In [160]:
df["rating"] = pd.to_numeric(df["rating"], errors="coerce")

In [161]:
df.rating_count = (
    df.rating_count
    .replace({"%": "", ",": ""}, regex=True)
    .astype(int)
)

In [162]:
df

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count
0,B07JW9H4J1,Wayona Nylon Braided USB to Lightning Fast Cha...,Computers&Accessories|Accessories&Peripherals|...,399.0,1099.0,64.0,4.2,24269
1,B098NS6PVG,Ambrane Unbreakable 60W / 3A Fast Charging 1.5...,Computers&Accessories|Accessories&Peripherals|...,199.0,349.0,43.0,4.0,43994
2,B096MSW6CT,Sounce Fast Phone Charging Cable & Data Sync U...,Computers&Accessories|Accessories&Peripherals|...,199.0,1899.0,90.0,3.9,7928
3,B08HDJ86NZ,boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...,Computers&Accessories|Accessories&Peripherals|...,329.0,699.0,53.0,4.2,94363
4,B08CF3B7N1,Portronics Konnect L 1.2M Fast Charging 3A 8 P...,Computers&Accessories|Accessories&Peripherals|...,154.0,399.0,61.0,4.2,16905
...,...,...,...,...,...,...,...,...
1460,B08L7J3T31,Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...,Home&Kitchen|Kitchen&HomeAppliances|WaterPurif...,379.0,919.0,59.0,4.0,1090
1461,B01M6453MB,Prestige Delight PRWO Electric Rice Cooker (1 ...,Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...,2280.0,3045.0,25.0,4.1,4118
1462,B009P2LIL4,Bajaj Majesty RX10 2000 Watts Heat Convector R...,"Home&Kitchen|Heating,Cooling&AirQuality|RoomHe...",2219.0,3080.0,28.0,3.6,468
1463,B00J5DYCCA,Havells Ventil Air DSP 230mm Exhaust Fan (Pist...,"Home&Kitchen|Heating,Cooling&AirQuality|Fans|E...",1399.0,1890.0,26.0,4.0,8031


In [163]:
df.isnull().sum()

product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 1
rating_count           0
dtype: int64

In [164]:
df = df.dropna()

## String parsing/splitting

In [165]:
## Category fix
df["category"] = df["category"].str.split("|").str[0]

C:\Users\Sahil Khan\AppData\Local\Temp\ipykernel_25548\454582929.py:2: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [166]:
df

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count
0,B07JW9H4J1,Wayona Nylon Braided USB to Lightning Fast Cha...,Computers&Accessories,399.0,1099.0,64.0,4.2,24269
1,B098NS6PVG,Ambrane Unbreakable 60W / 3A Fast Charging 1.5...,Computers&Accessories,199.0,349.0,43.0,4.0,43994
2,B096MSW6CT,Sounce Fast Phone Charging Cable & Data Sync U...,Computers&Accessories,199.0,1899.0,90.0,3.9,7928
3,B08HDJ86NZ,boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...,Computers&Accessories,329.0,699.0,53.0,4.2,94363
4,B08CF3B7N1,Portronics Konnect L 1.2M Fast Charging 3A 8 P...,Computers&Accessories,154.0,399.0,61.0,4.2,16905
...,...,...,...,...,...,...,...,...
1460,B08L7J3T31,Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...,Home&Kitchen,379.0,919.0,59.0,4.0,1090
1461,B01M6453MB,Prestige Delight PRWO Electric Rice Cooker (1 ...,Home&Kitchen,2280.0,3045.0,25.0,4.1,4118
1462,B009P2LIL4,Bajaj Majesty RX10 2000 Watts Heat Convector R...,Home&Kitchen,2219.0,3080.0,28.0,3.6,468
1463,B00J5DYCCA,Havells Ventil Air DSP 230mm Exhaust Fan (Pist...,Home&Kitchen,1399.0,1890.0,26.0,4.0,8031


# Data Transformation

## Feature engineering

In [167]:
df["Revenue_Potential"]=df["discounted_price"]*df["rating_count"]

C:\Users\Sahil Khan\AppData\Local\Temp\ipykernel_25548\1563695500.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



> **Note:** In this project, **Revenue Potential** is calculated as:
>
> **Revenue Potential = Discounted Price × Rating Count**


In [168]:
df

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count,Revenue_Potential
0,B07JW9H4J1,Wayona Nylon Braided USB to Lightning Fast Cha...,Computers&Accessories,399.0,1099.0,64.0,4.2,24269,9683331.0
1,B098NS6PVG,Ambrane Unbreakable 60W / 3A Fast Charging 1.5...,Computers&Accessories,199.0,349.0,43.0,4.0,43994,8754806.0
2,B096MSW6CT,Sounce Fast Phone Charging Cable & Data Sync U...,Computers&Accessories,199.0,1899.0,90.0,3.9,7928,1577672.0
3,B08HDJ86NZ,boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...,Computers&Accessories,329.0,699.0,53.0,4.2,94363,31045427.0
4,B08CF3B7N1,Portronics Konnect L 1.2M Fast Charging 3A 8 P...,Computers&Accessories,154.0,399.0,61.0,4.2,16905,2603370.0
...,...,...,...,...,...,...,...,...,...
1460,B08L7J3T31,Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...,Home&Kitchen,379.0,919.0,59.0,4.0,1090,413110.0
1461,B01M6453MB,Prestige Delight PRWO Electric Rice Cooker (1 ...,Home&Kitchen,2280.0,3045.0,25.0,4.1,4118,9389040.0
1462,B009P2LIL4,Bajaj Majesty RX10 2000 Watts Heat Convector R...,Home&Kitchen,2219.0,3080.0,28.0,3.6,468,1038492.0
1463,B00J5DYCCA,Havells Ventil Air DSP 230mm Exhaust Fan (Pist...,Home&Kitchen,1399.0,1890.0,26.0,4.0,8031,11235369.0


In [169]:
df['absolute_discounted_amount']=df['actual_price']-df['discounted_price']

C:\Users\Sahil Khan\AppData\Local\Temp\ipykernel_25548\1819317483.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [170]:
df

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count,Revenue_Potential,absolute_discounted_amount
0,B07JW9H4J1,Wayona Nylon Braided USB to Lightning Fast Cha...,Computers&Accessories,399.0,1099.0,64.0,4.2,24269,9683331.0,700.0
1,B098NS6PVG,Ambrane Unbreakable 60W / 3A Fast Charging 1.5...,Computers&Accessories,199.0,349.0,43.0,4.0,43994,8754806.0,150.0
2,B096MSW6CT,Sounce Fast Phone Charging Cable & Data Sync U...,Computers&Accessories,199.0,1899.0,90.0,3.9,7928,1577672.0,1700.0
3,B08HDJ86NZ,boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...,Computers&Accessories,329.0,699.0,53.0,4.2,94363,31045427.0,370.0
4,B08CF3B7N1,Portronics Konnect L 1.2M Fast Charging 3A 8 P...,Computers&Accessories,154.0,399.0,61.0,4.2,16905,2603370.0,245.0
...,...,...,...,...,...,...,...,...,...,...
1460,B08L7J3T31,Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...,Home&Kitchen,379.0,919.0,59.0,4.0,1090,413110.0,540.0
1461,B01M6453MB,Prestige Delight PRWO Electric Rice Cooker (1 ...,Home&Kitchen,2280.0,3045.0,25.0,4.1,4118,9389040.0,765.0
1462,B009P2LIL4,Bajaj Majesty RX10 2000 Watts Heat Convector R...,Home&Kitchen,2219.0,3080.0,28.0,3.6,468,1038492.0,861.0
1463,B00J5DYCCA,Havells Ventil Air DSP 230mm Exhaust Fan (Pist...,Home&Kitchen,1399.0,1890.0,26.0,4.0,8031,11235369.0,491.0


# Analysis

#### CO-Realtion

In [171]:
num_cols = [
    "discounted_price",
    "actual_price",
    "discount_percentage",
    "rating",
    "rating_count",
    "Revenue_Potential",
    "absolute_discounted_amount"
]

In [172]:

fig = px.scatter_matrix(
    df,
    dimensions=num_cols,
    title="Scatter Matrix of Numerical Features"
)

fig.update_layout(
    width=1500,    # Width in pixels
    height=1500    # Height in pixels
)

fig.show()

In [173]:
corr = df[num_cols].corr()

corr

,discounted_price,actual_price,discount_percentage,rating,rating_count,Revenue_Potential,absolute_discounted_amount
discounted_price,1.000000,0.961910,-0.242298,0.121132,-0.027304,0.468167,0.763650
actual_price,0.961910,1.000000,-0.117855,0.122467,-0.036215,0.413662,0.911056
discount_percentage,-0.242298,-0.117855,1.000000,-0.155679,0.011294,-0.193979,0.087079
rating,0.121132,0.122467,-0.155679,1.000000,0.102235,0.068720,0.106554
rating_count,-0.027304,-0.036215,0.011294,0.102235,1.000000,0.416633,-0.044354
Revenue_Potential,0.468167,0.413662,-0.193979,0.068720,0.416633,1.000000,0.270902
absolute_discounted_amount,0.763650,0.911056,0.087079,0.106554,-0.044354,0.270902,1.000000


# Correlation Analysis

## Objective

The purpose of the correlation analysis is to understand how different numerical variables relate to each other. Correlation values range from **-1 to +1**.

- **+1** → Perfect Positive Correlation
- **0** → No Correlation
- **-1** → Perfect Negative Correlation

Generally,

- **0.70 to 1.00** → Strong Positive
- **0.40 to 0.69** → Moderate Positive
- **0.10 to 0.39** → Weak Positive
- **-0.10 to -0.39** → Weak Negative
- **-0.40 to -0.69** → Moderate Negative
- **-0.70 to -1.00** → Strong Negative

---

# Key Findings

## 1. Actual Price vs Discounted Price
**Correlation = 0.962 (Very Strong Positive)**

### What does this mean?

Products with higher original prices also tend to have higher selling prices after discounts.

For example

| Actual Price | Discounted Price|
|--------------|-----------------|
| ₹500         | ₹450            |
| ₹50,000      | ₹42,000         |

Even after discounts, expensive products remain relatively expensive.

### Business Insight

- Pricing strategy is consistent.
- Discounts reduce prices proportionally instead of making premium products inexpensive.
- Premium products maintain their market positioning.

---

## 2. Discounted Price vs Absolute Discount Amount
**Correlation = 0.764 (Strong Positive)**

### What does this mean?

Higher-priced products generally receive larger discount amounts (in ₹).

Example

| Product | Discount |
|---------|----------|
| ₹500    | ₹50      |
| ₹50,000 | ₹10,000  |

Although both may have similar percentage discounts, the monetary savings are much larger for expensive products.

### Business Insight

- High-ticket products use larger monetary discounts to attract buyers.
- Businesses focus discount budgets on expensive products.

---

## 3. Actual Price vs Absolute Discount Amount
**Correlation = 0.911 (Very Strong Positive)**

### What does this mean?

The more expensive the product, the larger the discount amount offered.

Example

| Actual Price | Discount Amount |
|--------------|---------------- |
| ₹1,000       | ₹100            |
| ₹20,000      | ₹5,000          |

### Business Insight

Premium products rely heavily on discount campaigns to encourage purchases.

---

## 4. Discounted Price vs Revenue Potential
**Correlation = 0.468 (Moderate Positive)**

### What does this mean?

Higher selling prices contribute to higher revenue potential, but price alone does not determine revenue.

Revenue Potential is calculated as:

**Revenue Potential = Discounted Price × Rating Count**

Therefore, both price and customer demand influence revenue.

### Business Insight

Increasing price alone does not guarantee higher revenue. Products also need sufficient customer demand.

---

## 5. Rating Count vs Revenue Potential
**Correlation = 0.417 (Moderate Positive)**

### What does this mean?

Products with more customer reviews generally have higher revenue potential.

This is expected because Rating Count is one component of the Revenue Potential formula.

### Important Note

This correlation should not be interpreted as customer reviews causing higher revenue. Since Rating Count is part of the Revenue Potential calculation, a positive relationship is naturally expected.

### Business Insight

Products with strong customer engagement are likely to generate greater revenue opportunities.

---

## 6. Discount Percentage vs Discounted Price
**Correlation = -0.242 (Weak Negative)**

### What does this mean?

Products with larger percentage discounts often end up with lower selling prices.

However, the relationship is weak because pricing varies widely across different product categories.

### Business Insight

Percentage discounts alone do not explain product pricing. Original product price also plays an important role.

---

## 7. Discount Percentage vs Revenue Potential
**Correlation = -0.194 (Weak Negative)**

### What does this mean?

Higher discounts do not necessarily increase revenue potential.

Some products receive aggressive discounts but still have low demand.

### Business Insight

Large discounts alone are not enough to drive sales. Product quality, brand reputation, and customer demand are equally important.

---

## 8. Rating vs Revenue Potential
**Correlation = 0.069 (Almost No Correlation)**

### What does this mean?

Highly rated products do not automatically generate high revenue.

Many products have ratings between **4.0 and 4.5**, so ratings show little variation across products.

Revenue depends more on customer demand (Rating Count) and pricing than on rating itself.

### Business Insight

Businesses should focus on increasing product visibility and customer engagement instead of trying to improve ratings that are already high.

---

## 9. Rating vs Discount Percentage
**Correlation = -0.156 (Weak Negative)**

### What does this mean?

Products with larger discounts do not necessarily receive better customer ratings.

Customers rate products based primarily on quality, performance, and satisfaction—not discount size.

### Business Insight

Discounts can attract purchases but cannot compensate for poor product quality.

---

## 10. Rating Count vs Actual Price
**Correlation = -0.036 (No Meaningful Correlation)**

### What does this mean?

Expensive products do not consistently receive more or fewer reviews than cheaper products.

Customer engagement appears independent of product price.

### Business Insight

Businesses should not assume premium products naturally receive more customer attention. Marketing and product visibility remain important regardless of price.

---

# Overall Business Conclusion

The correlation analysis reveals that pricing variables are strongly related to each other, while customer satisfaction metrics (Rating and Rating Count) show relatively weak relationships with pricing.

Key observations include:

- Premium products remain premium even after discounts.
- Higher-priced products receive larger absolute discounts.
- Discounts alone do not guarantee higher revenue.
- Revenue potential is influenced by both selling price and customer demand.
- Product ratings have minimal impact on revenue compared to customer engagement.
- Increasing review counts and product visibility may provide greater business value than simply increasing discounts.

Overall, businesses should focus on **balanced pricing, stronger customer engagement, and review generation strategies rather than relying solely on discount campaigns to improve sales performance.**

In [174]:
# Finding
#.If actual price is more then there is more discounted price
#.Higher The actual price higher the discounted amount
#.The priduct which are in higher disount people buy them more


#### 1. Pricing Efficiency & Discount Strategy

###### 1.Are high discounts actually driving meaningful volume?

In [175]:
import plotly.express as px

fig = px.scatter(
    df,
    x="discount_percentage",
    y="rating_count",
    trendline="ols",
    title="Rating Count vs Discounted Price"
)

fig.show()

#### I suggests that higher discounts are not associated with proportionally higher customer engagement (using rating count as a proxy for demand). Therefore, discounting alone does not appear to be an effective strategy for increasing demand.

In [176]:
df.head()

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count,Revenue_Potential,absolute_discounted_amount
0,B07JW9H4J1,Wayona Nylon Braided USB to Lightning Fast Cha...,Computers&Accessories,399.0,1099.0,64.0,4.2,24269,9683331.0,700.0
1,B098NS6PVG,Ambrane Unbreakable 60W / 3A Fast Charging 1.5...,Computers&Accessories,199.0,349.0,43.0,4.0,43994,8754806.0,150.0
2,B096MSW6CT,Sounce Fast Phone Charging Cable & Data Sync U...,Computers&Accessories,199.0,1899.0,90.0,3.9,7928,1577672.0,1700.0
3,B08HDJ86NZ,boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...,Computers&Accessories,329.0,699.0,53.0,4.2,94363,31045427.0,370.0
4,B08CF3B7N1,Portronics Konnect L 1.2M Fast Charging 3A 8 P...,Computers&Accessories,154.0,399.0,61.0,4.2,16905,2603370.0,245.0


#### 2.Deep vs. Shallow Discount Performance:

In [177]:
df["discount_tier"] = pd.cut(
    df["discount_percentage"],
    bins=[0, 10, 30, 50, 100],
    labels=["0-10%", "10-30%", "30-50%", "50%+"],
    include_lowest=True
)

df["discount_tier"].value_counts()

C:\Users\Sahil Khan\AppData\Local\Temp\ipykernel_25548\3439198061.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



discount_tier
50%+      693
30-50%    426
10-30%    259
0-10%      84
Name: count, dtype: int64

In [178]:
tier_analysis = (
    df.groupby("discount_tier", observed=False)
      .agg(
          Avg_Rating_Count=("rating_count", "mean"),
          Avg_Revenue=("Revenue_Potential", "mean"),
          Avg_Discount=("discount_percentage", "mean"),
          Product_Count=("product_name", "count")
      )
      .reset_index()
)

# Convert to Thousands (K)
tier_analysis["Avg_Revenue"] = (tier_analysis["Avg_Revenue"] / 1000).round(2)

tier_analysis

,discount_tier,Avg_Rating_Count,Avg_Revenue,Avg_Discount,Product_Count
0,0-10%,12086.261905,55580.45,2.809524,84
1,10-30%,21457.169884,132503.10,22.166023,259
2,30-50%,16648.633803,51312.87,41.934272,426
3,50%+,18903.916306,15975.98,66.170274,693


In [179]:
comparison = tier_analysis.melt(
    id_vars="discount_tier",
    value_vars=["Avg_Rating_Count", "Avg_Revenue"],
    var_name="Metric",
    value_name="Value"
)

fig = px.bar(
    comparison,
    x="discount_tier",
    y="Value",
    color="Metric",
    barmode="group",
    title="Discount Tier Performance"
)

fig.update_layout(template="plotly_white")

fig.show()

##### 1.0–10% Discount Tier
###### Very low discounts generate reasonable revenue, but customer engagement remains lower than products receiving moderate discounts.

##### 2.10–30% Discount Tier ⭐ (Best Performing)
###### Products with moderate discounts (10–30%) achieve the best balance between customer demand and revenue generation. This appears to be the most effective pricing strategy.

##### 3.30–50% Discount Tier
###### Increasing discounts beyond 30% does not improve customer engagement and is associated with lower revenue potential.

##### 4.50%+ Discount Tier
###### Deep discounts (50%+) do not generate proportionally higher customer engagement and are associated with the lowest revenue potential, suggesting that excessive discounting may be reducing profitability.



# Why is 10–30% Better than 50%?

## Suppose 
Product A

Original Price = £100

Discount = 20%

Selling Price = £80

Rating Count = 2000

Revenue Potential:
80 × 2000
=160000

Product B

Original Price = £100

Discount = 60%

Selling Price = £40

Rating Count = 2300

Revenue Potential:
40 ×2300
=92000

See Customers Only increase 
2000 -> 23000 
But Price Decreasee 
80 -> 40 (Revenue almost half.) leads to (Over Discounting.)



## Product Distribution

Interesting Things.

Tier Products

1.0–10%	84

2.10–30%	259

3.30–50%	426

4.50%+	693

Almost Half Dataset Covers in 50% Discount Means Company Gives Discount Aggressively But The Product Above 50% Gives Least Reveneue so we need to improve Pricing strategy

# 2. Profitability & Revenue Realization

#### Discount  Leakage

##### Step 1: Total Discount Leakage

In [180]:
df["Total_Discount_Leakage"] = (
    df["absolute_discounted_amount"] * df["rating_count"]
)

C:\Users\Sahil Khan\AppData\Local\Temp\ipykernel_25548\1183686616.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



##### Step 2:Discount Leakage by Categorye

In [181]:
category_discount = (
    df.groupby("category")["Total_Discount_Leakage"]
      .sum()
      .sort_values(ascending=False)
      .reset_index()
)

# Convert to Lakhs
category_discount["Total_Discount_Leakage_Lakh"] = (
    category_discount["Total_Discount_Leakage"] / 100000
).round(2)

category_discount

,category,Total_Discount_Leakage,Total_Discount_Leakage_Lakh
0,Electronics,3.884105e+10,388410.53
1,Computers&Accessories,6.260240e+09,62602.40
2,Home&Kitchen,4.194989e+09,41949.89
3,MusicalInstruments,8.665899e+07,866.59
4,OfficeProducts,1.499209e+07,149.92
5,Health&PersonalCare,3.666663e+06,36.67
6,HomeImprovement,3.576244e+06,35.76
7,Car&Motorbike,1.856998e+06,18.57
8,Toys&Games,0.000000e+00,0.00


In [182]:

fig = px.bar(
    category_discount,
    x="category",
    y="Total_Discount_Leakage",
    title="Total Discount Leakage by Category",
    color="Total_Discount_Leakage",
    text_auto=".2s"
)

fig.update_layout(
    template="plotly_white",
    xaxis_title="Category",
    yaxis_title="Total Discount Amount (£)"
)

fig.show()

##### Step 3: Top 10 Products with Highest Discount Leakage

In [183]:
top_discount = (
    df.sort_values("Total_Discount_Leakage", ascending=False)
      .head(10)
)

top_discount

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count,Revenue_Potential,absolute_discounted_amount,discount_tier,Total_Discount_Leakage
568,B08VB57558,"Samsung Galaxy S20 FE 5G (Cloud Navy, 8GB RAM,...",Electronics,37990.0,74999.0,49.0,4.2,27790,1.055742e+09,37009.0,30-50%,1.028480e+09
935,B08GTYFC37,"SanDisk 1TB Extreme Portable SSD 1050MB/s R, 1...",Computers&Accessories,10389.0,32000.0,68.0,4.4,41398,4.300838e+08,21611.0,50%+,8.946522e+08
370,B09GFPVD9Y,"Redmi 9 Activ (Carbon Black, 4GB RAM, 64GB Sto...",Electronics,8499.0,10999.0,23.0,4.1,313836,2.667292e+09,2500.0,10-30%,7.845900e+08
72,B09RFC46VP,Redmi 108 cm (43 inches) 4K Ultra HD Android S...,Electronics,26999.0,42999.0,37.0,4.2,45238,1.221381e+09,16000.0,30-50%,7.238080e+08
371,B09GFLXVH9,"Redmi 9A Sport (Coral Green, 2GB RAM, 32GB Sto...",Electronics,6499.0,8499.0,24.0,4.1,313836,2.039620e+09,2000.0,10-30%,6.276720e+08
566,B09GFPN6TP,"Redmi 9A Sport (Coral Green, 3GB RAM, 32GB Sto...",Electronics,7499.0,9499.0,21.0,4.1,313832,2.353426e+09,2000.0,10-30%,6.276640e+08
124,B08Y55LPBF,Redmi 126 cm (50 inches) 4K Ultra HD Android S...,Electronics,32999.0,44999.0,27.0,4.2,45238,1.492809e+09,12000.0,10-30%,5.428560e+08
64,B09F9YQQ7B,Redmi 80 cm (32 inches) Android 11 Series HD R...,Electronics,13999.0,24999.0,44.0,4.2,45238,6.332868e+08,11000.0,30-50%,4.976180e+08
953,B09F9YQQ7B,Redmi 80 cm (32 inches) Android 11 Series HD R...,Electronics,13999.0,24999.0,44.0,4.2,45237,6.332728e+08,11000.0,30-50%,4.976070e+08
332,B09RWQ7YR6,MI 138.8 cm (55 inches) 5X Series 4K Ultra HD ...,Electronics,46999.0,69999.0,33.0,4.3,21252,9.988227e+08,23000.0,30-50%,4.887960e+08


##### Step 4: Compare Discount Lekage vs Revenue (Category)

In [184]:
category_compare = (
    df.groupby("category")
      .agg(
          Total_Discount_Leakage=("Total_Discount_Leakage","sum"),
          Revenue_Potential=("Revenue_Potential","sum")
      )
      .reset_index()
)

comparison = category_compare.melt(
    id_vars="category",
    value_vars=["Total_Discount_Leakage","Revenue_Potential"],
    var_name="Metric",
    value_name="Value"
)

fig = px.bar(
    comparison,
    x="category",
    y="Value",
    color="Metric",
    barmode="group",
    text_auto=".2s"
)

fig.update_layout(height=600, width=1200)
fig.show()

### Electronics  (Largest Category)

##### Electronics is the company's largest revenue-generating category. Although it receives the highest discounts, it also generates the highest revenue potential, indicating that discounts in this category are likely driving substantial customer demand.

###### Electronics
39 Billion Discount -> 59 Billion Revenue
Means For Every 1 Rs Discount Amazon Genrate 59/38 -> 1.5 Rs



### Computers & Accessories

##### The category shows a relatively balanced relationship between discount leakage and revenue potential. Discounts appear to be contributing to demand, but their efficiency should be monitored to ensure profitability.


### Home & Kitchen

##### Home & Kitchen delivers stronger revenue relative to the discounts offered, suggesting that promotional pricing in this category is comparatively efficient.

### Health & Personal Care

##### The category appears to receive relatively high discounts without generating proportionally higher revenue. This may indicate an opportunity to review pricing or promotional strategies.



#### Insight 
The Electronics category contributes both the highest estimated discount leakage (£39B) and the highest revenue potential (£59B), suggesting that promotional pricing in this category is associated with strong customer demand. In contrast, categories such as Health & Personal Care exhibit relatively high discount leakage compared to revenue potential, indicating that discounts may not be generating proportional returns. This suggests that the company should review discount strategies for lower-performing categories while maintaining optimized promotions in high-performing ones.


##### Step 5: Discount Efficiency Ratio 

In [185]:
df["Discount_Efficiency"] = (
    df["Revenue_Potential"] /
    df["Total_Discount_Leakage"]
).round(2)

C:\Users\Sahil Khan\AppData\Local\Temp\ipykernel_25548\924716093.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [186]:
lowest_efficiency = (
    df.sort_values("Discount_Efficiency")
      .loc[:, [
          "product_id",
          "product_name",
          "category",
          'actual_price',
          "absolute_discounted_amount",
          "Revenue_Potential",
        "Total_Discount_Leakage",
          "Discount_Efficiency"
      ]]
      .head(10)
)

lowest_efficiency

,product_id,product_name,category,actual_price,absolute_discounted_amount,Revenue_Potential,Total_Discount_Leakage,Discount_Efficiency
695,B097C564GC,"rts [2 Pack] Mini USB C Type C Adapter Plug, T...",Computers&Accessories,4999.0,4705.0,1301244.0,20824330.0,0.06
372,B0BF4YBLPX,"Fire-Boltt Ninja Call Pro Plus 1.83"" Smart Wat...",Electronics,19999.0,18200.0,25072663.0,253653400.0,0.10
364,B0BF54972T,"Fire-Boltt Ninja Call Pro Plus 1.83"" Smart Wat...",Electronics,19999.0,18200.0,25072663.0,253653400.0,0.10
368,B0BF563HB4,"Fire-Boltt Ninja Call Pro Plus 1.83"" Smart Wat...",Electronics,19999.0,18200.0,25072663.0,253653400.0,0.10
380,B0BF54LXW6,"Fire-Boltt Ninja Call Pro Plus 1.83"" Smart Wat...",Electronics,19999.0,18200.0,25072663.0,253653400.0,0.10
334,B0BF57RN3K,"Fire-Boltt Ninja Call Pro Plus 1.83"" Smart Wat...",Electronics,19999.0,18200.0,25072663.0,253653400.0,0.10
644,B085HY1DGR,Sounce Spiral Charger Cable Protector Data Cab...,Computers&Accessories,999.0,900.0,138204.0,1256400.0,0.11
407,B085HY1DGR,Sounce Spiral Charger Cable Protector Data Cab...,Computers&Accessories,999.0,900.0,138204.0,1256400.0,0.11
920,B08X77LM8C,"Silicone Rubber Earbuds Tips, Eartips, Earpads...",Electronics,999.0,900.0,58806.0,534600.0,0.11
602,B08W56G1K9,LAPSTER Spiral Charger Spiral Charger Cable Pr...,Computers&Accessories,999.0,900.0,866349.0,7875900.0,0.11


## Lowest Discount Efficiency Products

### 1. Product **B097C564GC** has the Lowest Discount Efficiency (**0.06**)

- **Actual Price:** £4,999
- **Discount Given:** £4,705 (≈94% Discount)
- **Discount Efficiency:** **0.06**

#### Key Insight

- The company is offering an extremely high discount (**approximately 94%**).
- For every **£1 of discount leakage**, the product generates only **£0.06 of Revenue Potential**, indicating a highly inefficient pricing strategy.
- Such aggressive discounting significantly reduces pricing efficiency.

### Business Recommendations

- Review products with extremely low Discount Efficiency.
- Avoid offering excessively high discounts unless they are supported by strong sales growth.
- Re-evaluate the pricing and promotional strategy for these products.

---

## 2. Fire-Boltt Smart Watch Products (Discount Efficiency = **0.10**)

### Observation

The same Fire-Boltt Smart Watch appears multiple times with different Product IDs but very similar pricing and discount values.

### Key Metrics

- **Revenue Potential:** ≈ **£25M**
- **Total Discount Leakage:** ≈ **£254M**

### Business Insight

This indicates that the company is providing very large discounts relative to the selling price.

There are two possible explanations:

1. If these are **duplicate listings**, the same pricing strategy has been applied across multiple listings.
2. If these are **different SKUs or product variants**, the entire Fire-Boltt product line is being heavily discounted.

Both scenarios suggest that the current pricing strategy should be reviewed to improve profitability.

---

## Overall Business Recommendations

###  Review Heavy Discount Products

- Identify products with very low Discount Efficiency.
- Evaluate whether the current discount levels are justified by the revenue generated.

### Reduce Aggressive Discounting

- Avoid extremely high discounts (e.g., above 80–90%) unless they clearly improve business performance.
- Consider implementing a maximum discount threshold to protect profit margins.

### Focus on Moderate Discounts

From the previous **Discount Tier Performance** analysis, the **10–30% discount tier** generated the strongest Revenue Potential.

This suggests that:

- Moderate discounts are more sustainable.
- Deep discounts do not necessarily generate proportionally higher revenue.
- Optimising discounts within the **10–30% range** may improve long-term profitability while reducing unnecessary discount leakage. 

In [187]:
df

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count,Revenue_Potential,absolute_discounted_amount,discount_tier,Total_Discount_Leakage,Discount_Efficiency
0,B07JW9H4J1,Wayona Nylon Braided USB to Lightning Fast Cha...,Computers&Accessories,399.0,1099.0,64.0,4.2,24269,9683331.0,700.0,50%+,16988300.0,0.57
1,B098NS6PVG,Ambrane Unbreakable 60W / 3A Fast Charging 1.5...,Computers&Accessories,199.0,349.0,43.0,4.0,43994,8754806.0,150.0,30-50%,6599100.0,1.33
2,B096MSW6CT,Sounce Fast Phone Charging Cable & Data Sync U...,Computers&Accessories,199.0,1899.0,90.0,3.9,7928,1577672.0,1700.0,50%+,13477600.0,0.12
3,B08HDJ86NZ,boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...,Computers&Accessories,329.0,699.0,53.0,4.2,94363,31045427.0,370.0,50%+,34914310.0,0.89
4,B08CF3B7N1,Portronics Konnect L 1.2M Fast Charging 3A 8 P...,Computers&Accessories,154.0,399.0,61.0,4.2,16905,2603370.0,245.0,50%+,4141725.0,0.63
...,...,...,...,...,...,...,...,...,...,...,...,...,...
1460,B08L7J3T31,Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...,Home&Kitchen,379.0,919.0,59.0,4.0,1090,413110.0,540.0,50%+,588600.0,0.70
1461,B01M6453MB,Prestige Delight PRWO Electric Rice Cooker (1 ...,Home&Kitchen,2280.0,3045.0,25.0,4.1,4118,9389040.0,765.0,10-30%,3150270.0,2.98
1462,B009P2LIL4,Bajaj Majesty RX10 2000 Watts Heat Convector R...,Home&Kitchen,2219.0,3080.0,28.0,3.6,468,1038492.0,861.0,10-30%,402948.0,2.58
1463,B00J5DYCCA,Havells Ventil Air DSP 230mm Exhaust Fan (Pist...,Home&Kitchen,1399.0,1890.0,26.0,4.0,8031,11235369.0,491.0,10-30%,3943221.0,2.85


# Price Realization Ratio:

In [188]:
df["Price_Realization_Ratio"] = (
    df["discounted_price"] /
    df["actual_price"]
)

C:\Users\Sahil Khan\AppData\Local\Temp\ipykernel_25548\2379357509.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [189]:
df["Price_Realization_%"] = (
    df["discounted_price"] /
    df["actual_price"]
) * 100

C:\Users\Sahil Khan\AppData\Local\Temp\ipykernel_25548\345607392.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [190]:
price_realization = (
    df.groupby("category")
      .agg(
          Avg_Realization=("Price_Realization_Ratio","mean"),
          Avg_Discount=("discount_percentage","mean"),
          Product_Count=("product_name","count")
      )
      .reset_index()
)

price_realization

,category,Avg_Realization,Avg_Discount,Product_Count
0,Car&Motorbike,0.584750,42.000000,1
1,Computers&Accessories,0.460818,53.920177,451
2,Electronics,0.491846,50.828897,526
3,Health&PersonalCare,0.473158,53.000000,1
4,Home&Kitchen,0.598292,40.174497,447
5,HomeImprovement,0.420559,57.500000,2
6,MusicalInstruments,0.541917,46.000000,2
7,OfficeProducts,0.876402,12.354839,31
8,Toys&Games,1.000000,0.000000,1


# 📊 Price Realization Ratio Analysis

## What is Price Realization Ratio?

The **Price Realization Ratio** measures how much of the original product price the company actually receives after applying discounts.

### Formula

Price Realization Ratio}=Discounted Price/Actual Price


A **higher ratio** indicates that the company preserves more of its original selling price, while a **lower ratio** means the company depends heavily on discounts to drive sales.

---

# What does "Preserve Price Integrity" mean?

**Price Integrity** means customers are willing to purchase products close to their original listed price without requiring large discounts.

### Example

- **Original Price:** £1000
- **Selling Price:** £950

**Price Realization Ratio = 0.95**

This means the company keeps **95% of the original selling price**, indicating strong pricing power and healthy profit margins.

Categories with high Price Realization Ratios generally have:

- Strong customer demand
- Better profit margins
- Lower dependence on discounts
- Higher perceived product value

---

# What does "Rely Heavily on Deep Discounting" mean?

When the Price Realization Ratio is low, the company must significantly reduce prices to generate sales.

### Example

- **Original Price:** £1000
- **Selling Price:** £350

**Price Realization Ratio = 0.35**

This means the company receives only **35% of the original selling price**.

Such categories generally indicate:

- Heavy promotional pricing
- Lower profit margins
- Intense market competition
- Inventory clearance strategies

---

# Business Insights

### 1. Office Products has the highest Price Realization Ratio (0.88)

- Customers pay nearly **88% of the original product price**.
- Average discount is only **12.35%**, the lowest among major categories.
- This category demonstrates **excellent price integrity** and strong pricing power.
- The company does not rely on aggressive discounts to generate sales.

---

### 2. Home & Kitchen performs well (0.60)

- Customers pay approximately **60% of the original price**.
- Average discount is around **40%**, which is moderate compared to other categories.
- This suggests discounts are sufficient to attract customers while maintaining reasonable profit margins.

---

### 3. Car & Motorbike shows a good realization ratio (0.58)

- The category retains around **58% of its original price**.
- However, this result is based on **only one product**, so it should not be used to make broad business decisions.

---

### 4. Musical Instruments also maintains healthy pricing (0.54)

- Customers pay approximately **54% of the original price**.
- Average discounts remain moderate.
- Since the category contains only **two products**, further data would be required before drawing strong conclusions.

---

### 5. Electronics relies heavily on discounting (0.49)

- Customers pay only **49% of the original selling price**.
- Average discount exceeds **50%**.
- Despite being one of the largest product categories (526 products), the company heavily depends on discounts to generate sales.
- This may indicate intense competition or an aggressive promotional strategy.

---

### 6. Computers & Accessories also depends on discounts (0.46)

- Customers pay only **46% of the original product price**.
- Average discount is approximately **54%**.
- This category contains **451 products**, making it one of the largest segments.
- Large discounts suggest weaker pricing power and potential pressure on profitability.

---

### 7. Home Improvement has the lowest realization ratio among multi-product categories (0.42)

- Customers pay only **42% of the original price**.
- Average discount is **57.5%**, the highest among major categories.
- The company appears to rely heavily on price reductions to drive sales.
- However, since the category contains only **two products**, this insight should be interpreted cautiously.

---

### 8. Toys & Games shows a realization ratio of 1.00

- Products are sold at their original listed price.
- No discount is provided.
- However, the category contains only **one product**, so this result is not statistically meaningful.

---

# Overall Business Insight

The analysis reveals a clear relationship between discounting and price realization.

- **Office Products** preserves the highest price integrity by selling products close to their original prices with minimal discounts.
- **Home & Kitchen** maintains a healthy balance between discounts and revenue realization.
- **Electronics** and **Computers & Accessories** rely heavily on deep discounting, with customers paying less than half of the original selling price on average.
- Categories with fewer than five products (Car & Motorbike, Toys & Games, Musical Instruments, Home Improvement, Health & Personal Care) should be interpreted carefully because the sample size is too small for reliable business conclusions.

---

# Business Recommendations

### ✅ Maintain Premium Pricing

Continue the current pricing strategy for categories such as **Office Products**, where customers are willing to purchase products close to their original prices.

---

### ✅ Review Electronics Pricing Strategy

Electronics and Computers & Accessories should be reviewed to determine whether discounts above **50%** are genuinely required or whether smaller discounts can achieve similar sales.

---

### ✅ Focus on Moderate Discounts

Previous analysis showed that the **10–30% discount tier** generated the highest Revenue Potential.

Instead of offering discounts above **50%**, the company should experiment with moderate discounts that can:

- Improve profit margins
- Reduce discount leakage
- Maintain sustainable long-term revenue

---

### ✅ Increase Price Integrity

For categories with low Price Realization Ratios, the company should focus on:

- Stronger product positioning
- Better marketing campaigns
- Value-added bundles
- Loyalty programs

rather than relying solely on deep discounts.

---



# 3.Product Portfolio & Category Dynamics

## Category Revenue Matrix

In [191]:
category_matrix = (
    df.groupby("category")
      .agg(
          Avg_Selling_Price=("discounted_price", "mean"),
          Total_Revenue=("Revenue_Potential", "sum"),
          Product_Count=("product_name", "count")
      )
      .reset_index()
)

category_matrix

,category,Avg_Selling_Price,Total_Revenue,Product_Count
0,Car&Motorbike,2339.000000,2.615002e+06,1
1,Computers&Accessories,845.393836,6.354569e+09,451
2,Electronics,5965.887833,5.917975e+10,526
3,Health&PersonalCare,899.000000,3.293037e+06,1
4,Home&Kitchen,2331.133803,6.262254e+09,447
5,HomeImprovement,337.000000,2.587190e+06,2
6,MusicalInstruments,638.000000,6.445808e+07,2
7,OfficeProducts,301.580645,4.578673e+07,31
8,Toys&Games,150.000000,2.380050e+06,1


In [192]:

fig = px.scatter(
    category_matrix,
    x="Avg_Selling_Price",
    y="Total_Revenue",
    size="Product_Count",
    color="category",
    hover_name="category",

    title="Category Revenue Matrix (BCG Style)",
    size_max=60
)

fig.update_traces(textposition="top center")

fig.update_layout(
    template="plotly_white",
    width=1300,
    height=700,

    legend=dict(
        title=dict(text="Category"),
        orientation="v",
        yanchor="top",
        y=1,
        xanchor="left",
        x=1.02
    ),

    margin=dict(r=220)
)

fig.show()

# Category Revenue Matrix (BCG Style) – Business Insights

## Objective
The Category Revenue Matrix helps identify which product categories contribute the most revenue, which categories command premium prices, and which categories have limited business impact.

**Chart Interpretation**
- **X-axis:** Average Selling Price (£)
- **Y-axis:** Total Revenue Potential
- **Bubble Size:** Number of Products in the Category

---

# Key Business Insights

## 1. Electronics is the Market Leader 

### Observation
- Electronics has the **highest average selling price (~£6,000)**.
- It also generates the **highest revenue potential (~£59B)**.
- The bubble is the largest, indicating a large product portfolio.

### Business Insight
Electronics is the company's flagship category. Customers are willing to purchase high-value products, and despite premium pricing, this category contributes the largest share of revenue.

### Recommendation
- Continue investing in Electronics.
- Prioritize inventory availability.
- Introduce premium bundles and extended warranties.
- Focus marketing efforts on this category since every additional sale generates significant revenue.

---

## 2. Home & Kitchen is a Strong Revenue Generator 

### Observation
- Average selling price is around **£2,300**.
- Revenue potential is approximately **£6B**.
- Large product portfolio.

### Business Insight
Although products are much cheaper than Electronics, the category still generates substantial revenue through higher sales volume.

### Recommendation
- Maintain competitive pricing.
- Use seasonal promotions.
- Increase cross-selling opportunities with related household products.

---

## 3. Computers & Accessories is a High-Performing Category 

### Observation
- Average selling price is around **£800–900**.
- Revenue potential is approximately **£6B**.

### Business Insight
The category balances affordability with strong sales volume. It attracts a broad customer base and serves as a reliable revenue contributor.

### Recommendation
- Bundle accessories with laptops and PCs.
- Promote add-on purchases such as keyboards, mice, and storage devices.
- Focus on upselling premium accessories.

---

## 4. Office Products has Moderate Business Impact 

### Observation
- Moderate selling price.
- Revenue is significantly lower than the top three categories.

### Business Insight
The category contributes steady revenue but is not a primary business driver.

### Recommendation
- Optimize inventory.
- Improve product assortment.
- Target business customers through B2B promotions.

---

## 5. Small Categories Contribute Minimal Revenue 

Categories include:
- Car & Motorbike
- Home Improvement
- Health & Personal Care
- Musical Instruments
- Toys & Games

### Observation
- Low average selling prices.
- Very low total revenue.
- Small bubble sizes indicate limited product portfolios.

### Business Insight
These categories currently have a relatively small impact on overall business performance.

### Recommendation
- Evaluate profitability before expanding inventory.
- Focus marketing only if margins justify investment.
- Consider removing consistently underperforming products.

---

# Portfolio Classification (BCG Style)

## Star
- Electronics

High selling price + High revenue

**Action:** Continue investing and expanding.

---

## Cash Cows
- Home & Kitchen
- Computers & Accessories

Moderate prices with strong revenue.

**Action:** Maintain pricing and maximize profitability.

---

## Question Marks
- Office Products

Shows growth potential but requires better marketing and product optimization.

---

## Low Performers
- Toys & Games
- Musical Instruments
- Home Improvement
- Car & Motorbike
- Health & Personal Care

Limited contribution to overall revenue.

**Action:** Review inventory, optimize pricing, and evaluate whether additional investment is justified.

---

# Overall Business Conclusion

The company's revenue is highly concentrated in **Electronics**, making it the primary revenue engine. **Home & Kitchen** and **Computers & Accessories** provide stable secondary revenue streams. The remaining categories contribute only a small portion of total revenue and should be reviewed to determine whether inventory expansion or promotional spending is justified.

From a strategic perspective, the business should continue investing in its top-performing categories while optimizing or rationalizing low-performing categories to improve overall profitability.

## Top & Bottom Performers (Pareto Principle (80/20 Rule).)

In [193]:
## Step 1: Sort products by Revenue Potential
pareto_df = (
    df.sort_values("Revenue_Potential", ascending=False)
      .reset_index(drop=True)
)

In [194]:
## Step 2: Calculate Total Revenue Potential
total_revenue = pareto_df["Revenue_Potential"].sum()

In [195]:
## Step 3: Calculate Cumulative Revenue
pareto_df["Cumulative_Revenue"] = pareto_df["Revenue_Potential"].cumsum()

In [196]:
## Step 4: Calculate Cumulative Percentage
pareto_df["Cumulative_%"] = (
    pareto_df["Cumulative_Revenue"] / total_revenue
) * 100

In [197]:
## Step 5: Calculate Product Percentage
pareto_df["Product_%"] = (
    (pareto_df.index + 1) / len(pareto_df)
) * 100

In [198]:
## Step 6: Find Products Responsible for 80% Revenue
top_products = pareto_df[pareto_df["Cumulative_%"] <= 80]

top_products.head()

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count,Revenue_Potential,absolute_discounted_amount,discount_tier,Total_Discount_Leakage,Discount_Efficiency,Price_Realization_Ratio,Price_Realization_%,Cumulative_Revenue,Cumulative_%,Product_%
0,B09GFPVD9Y,"Redmi 9 Activ (Carbon Black, 4GB RAM, 64GB Sto...",Electronics,8499.0,10999.0,23.0,4.1,313836,2.667292e+09,2500.0,10-30%,784590000.0,3.40,0.772707,77.270661,2.667292e+09,3.708812,0.068399
1,B09GFPN6TP,"Redmi 9A Sport (Coral Green, 3GB RAM, 32GB Sto...",Electronics,7499.0,9499.0,21.0,4.1,313832,2.353426e+09,2000.0,10-30%,627664000.0,3.75,0.789452,78.945152,5.020718e+09,6.981200,0.136799
2,B09GFLXVH9,"Redmi 9A Sport (Coral Green, 2GB RAM, 32GB Sto...",Electronics,6499.0,8499.0,24.0,4.1,313836,2.039620e+09,2000.0,10-30%,627672000.0,3.25,0.764678,76.467820,7.060338e+09,9.817248,0.205198
3,B09GFM8CGS,"Redmi 9A Sport (Carbon Black, 2GB RAM, 32GB St...",Electronics,6499.0,7999.0,19.0,4.1,313832,2.039594e+09,1500.0,10-30%,470748000.0,4.33,0.812477,81.247656,9.099933e+09,12.653259,0.273598
4,B08Y55LPBF,Redmi 126 cm (50 inches) 4K Ultra HD Android S...,Electronics,32999.0,44999.0,27.0,4.2,45238,1.492809e+09,12000.0,10-30%,542856000.0,2.75,0.733327,73.332741,1.059274e+10,14.728978,0.341997


In [199]:
## Step 7: Top 20 Products
num_products = len(top_products)

percentage_products = (
    num_products / len(df)
) * 100

print(f"{num_products} products generate approximately 80% of Revenue Potential.")
print(f"That is {percentage_products:.2f}% of all products.")

168 products generate approximately 80% of Revenue Potential.
That is 11.49% of all products.


In [200]:
## Step 8: Bottom 20 Products
top20 = pareto_df[[
    "product_name",
    "category",
    "actual_price",
    "discounted_price",
    "rating_count",
    "Revenue_Potential"
]].head(20)

top20

,product_name,category,actual_price,discounted_price,rating_count,Revenue_Potential
0,"Redmi 9 Activ (Carbon Black, 4GB RAM, 64GB Sto...",Electronics,10999.0,8499.0,313836,2.667292e+09
1,"Redmi 9A Sport (Coral Green, 3GB RAM, 32GB Sto...",Electronics,9499.0,7499.0,313832,2.353426e+09
2,"Redmi 9A Sport (Coral Green, 2GB RAM, 32GB Sto...",Electronics,8499.0,6499.0,313836,2.039620e+09
3,"Redmi 9A Sport (Carbon Black, 2GB RAM, 32GB St...",Electronics,7999.0,6499.0,313832,2.039594e+09
4,Redmi 126 cm (50 inches) 4K Ultra HD Android S...,Electronics,44999.0,32999.0,45238,1.492809e+09
5,Redmi 108 cm (43 inches) 4K Ultra HD Android S...,Electronics,42999.0,26999.0,45238,1.221381e+09
6,"Samsung Galaxy S20 FE 5G (Cloud Navy, 8GB RAM,...",Electronics,74999.0,37990.0,27790,1.055742e+09
7,MI 138.8 cm (55 inches) 5X Series 4K Ultra HD ...,Electronics,69999.0,46999.0,21252,9.988227e+08
8,OnePlus 108 cm (43 inches) Y Series Full HD Sm...,Electronics,31999.0,24999.0,34899,8.724401e+08
9,MI 108 cm (43 inches) 5A Series Full HD Smart ...,Electronics,35999.0,24999.0,32840,8.209672e+08


In [201]:
## Step 9: Underperforming Premium Products
underperformers = (
    df[
        (df["actual_price"] > df["actual_price"].quantile(0.75)) &
        (df["Revenue_Potential"] < df["Revenue_Potential"].quantile(0.25))
    ]
    .sort_values("Revenue_Potential")
)

underperformers.head(20)

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count,Revenue_Potential,absolute_discounted_amount,discount_tier,Total_Discount_Leakage,Discount_Efficiency,Price_Realization_Ratio,Price_Realization_%
1378,B08T8KWNQ9,TE™ Instant Electric Heating Hot and Cold Wate...,Home&Kitchen,1449.0,4999.0,71.0,3.6,63,91287.0,3550.0,50%+,223650.0,0.41,0.289858,28.985797
299,B0BNDD9TN6,WANBO X1 Pro (Upgraded) | Native 1080P Full HD...,Electronics,13990.0,28900.0,52.0,4.5,7,97930.0,14910.0,50%+,104370.0,0.94,0.484083,48.408304
487,B0BGSV43WY,Noise ColorFit Pro 4 Alpha Bluetooth Calling S...,Electronics,4499.0,7999.0,44.0,3.5,37,166463.0,3500.0,30-50%,129500.0,1.29,0.562445,56.244531
210,B0BLV1GNLN,WZATCO Pixel | Portable LED Projector | Native...,Electronics,6490.0,9990.0,35.0,4.0,27,175230.0,3500.0,30-50%,94500.0,1.85,0.649650,64.964965
402,B0BP18W8TM,"Fire-Boltt Gladiator 1.96"" Biggest Display Sma...",Electronics,3999.0,9999.0,60.0,4.4,73,291927.0,6000.0,50%+,438000.0,0.67,0.399940,39.993999
298,B09HN7LD5L,PROLEGEND® PL-T002 Universal TV Stand Table To...,Electronics,1850.0,4500.0,59.0,4.0,184,340400.0,2650.0,50%+,487600.0,0.70,0.411111,41.111111
525,B0BNVBJW2S,"boAt Newly Launched Wave Electra with 1.81"" HD...",Electronics,2499.0,7990.0,69.0,4.1,154,384846.0,5491.0,50%+,845614.0,0.46,0.312766,31.276596
462,B0BNV7JM5Y,"boAt Newly Launched Wave Electra with 1.81"" HD...",Electronics,2999.0,7990.0,62.0,4.1,154,461846.0,4991.0,50%+,768614.0,0.60,0.375344,37.534418
512,B0BMVWKZ8G,"Newly Launched Boult Dive+ with 1.85"" HD Displ...",Electronics,1999.0,8499.0,76.0,4.3,240,479760.0,6500.0,50%+,1560000.0,0.31,0.235204,23.520414
1054,B0B8XNPQPN,"Pigeon Healthifry Digital Air Fryer, 360° High...",Home&Kitchen,3599.0,7950.0,55.0,4.2,136,489464.0,4351.0,50%+,591736.0,0.83,0.452704,45.270440


In [202]:
## Step 10: Pareto Chart (Highly Recommended)
import plotly.graph_objects as go

fig = go.Figure()

# Revenue Potential Bars
fig.add_trace(go.Bar(
    x=pareto_df.index + 1,
    y=pareto_df["Revenue_Potential"],
    name="Revenue Potential"
))

# Cumulative Percentage Line
fig.add_trace(go.Scatter(
    x=pareto_df.index + 1,
    y=pareto_df["Cumulative_%"],
    mode="lines",
    name="Cumulative %",
    yaxis="y2",
    line=dict(color="red", width=3)
))

fig.update_layout(
    title="Pareto Analysis (80/20 Rule)",
    xaxis_title="Products (Sorted by Revenue Potential)",
    yaxis_title="Revenue Potential",

    yaxis2=dict(
        title="Cumulative %",
        overlaying="y",
        side="right",
        range=[0, 100]
    ),

    template="plotly_white",
    width=1200,
    height=650
)

fig.show()

# Pareto Analysis (80/20 Rule) - Business Insights

## Objective

The Pareto Principle (80/20 Rule) helps identify the small number of products that generate the majority of business value. In this analysis, Revenue Potential is calculated as:

**Revenue Potential = Discounted Price × Rating Count**

Products are sorted in descending order of Revenue Potential, and cumulative contribution is analyzed.

---

# Executive Summary

### Key Finding

- Only **168 products** out of **1,465 total products** contribute approximately **80% of the total Revenue Potential**.
- This represents only **11.49%** of the entire product catalog.

> **Business Insight:** A very small portion of products drives most of the business value, which is an even stronger concentration than the traditional 80/20 rule.

---

# 1. Revenue Distribution is Highly Uneven

The Pareto curve rises very steeply in the beginning and then gradually flattens.

This indicates that:

- A handful of products generate extremely high Revenue Potential.
- Most remaining products contribute very little individually.

This is a classic **long-tail distribution**, commonly seen in e-commerce businesses.

### Business Meaning

Instead of treating all products equally, the company should prioritize the products that generate the majority of revenue.

---

# 2. Top Revenue Drivers

The Top 20 products are dominated by the **Electronics** category.

Examples include:

- Redmi Smartphones
- Samsung Smartphones
- Mi Smart TVs
- OnePlus Smart TVs

These products have:

- High Rating Count
- Competitive Selling Price
- Strong Customer Demand

As a result, they generate exceptionally high Revenue Potential.

### Business Recommendation

These products should receive:

- Higher inventory allocation
- Priority advertising budget
- Homepage placement
- Better delivery availability
- Promotional campaigns

Since these products generate most of the revenue, stock shortages should be avoided.

---

# 3. Electronics is the Growth Engine

Almost every product in the Top 20 belongs to Electronics.

This indicates that Electronics is the primary revenue-generating category.

Possible reasons include:

- Higher selling prices
- Large customer demand
- High purchase frequency
- Strong customer trust

### Business Recommendation

Increase investment in:

- Smartphones
- Smart TVs
- Electronics accessories

Launching more products in this category is likely to increase total revenue.

---

# 4. Large Number of Products Generate Very Little Revenue

After approximately the first 170 products, the Pareto curve becomes almost flat.

This means:

Adding hundreds of additional products contributes very little to total Revenue Potential.

### Business Meaning

Maintaining these products may increase:

- Inventory costs
- Warehousing costs
- Operational complexity

while providing very limited business value.

---

# 5. Underperforming Premium Products

Several expensive products appear in the Underperforming Products table.

Characteristics include:

- High Actual Price
- Low Rating Count
- Low Revenue Potential

Examples include:

- Premium projectors
- Premium smart watches
- Premium televisions
- Kitchen appliances

These products are expensive but have very limited customer demand.

### Possible Reasons

- Poor visibility
- Weak marketing
- High pricing
- Strong competition
- Low customer awareness

---

# 6. Premium Pricing Alone Does Not Guarantee Revenue

Several expensive products have very low Revenue Potential despite costing thousands of rupees.

Revenue depends on:

Revenue Potential = Price × Demand

If customer demand is low, even expensive products contribute very little.

### Business Lesson

Increasing prices alone will not increase business revenue.

Demand is equally important.

---

# 7. Importance of Rating Count

The top-performing products all have extremely high Rating Counts.

Examples:

- 300,000+ customer ratings
- 50,000+ customer ratings

High Rating Count indicates:

- Strong customer trust
- High popularity
- High purchase frequency

These products naturally generate higher Revenue Potential.

### Business Recommendation

Encourage customer reviews through:

- Cashback
- Reward points
- Email reminders
- Review incentives

More reviews improve customer confidence and increase future sales.

---

# 8. Inventory Optimization Opportunity

Since only 11.49% of products generate around 80% of Revenue Potential, inventory investment should focus primarily on these products.

### Business Recommendation

Maintain higher inventory for:

- Top-performing products

Reduce inventory for:

- Slow-moving products
- Low-demand premium products

This reduces warehouse costs while improving product availability.

---

# 9. Marketing Budget Optimization

Instead of equally promoting every product, marketing resources should prioritize high-performing products.

Recommended allocation:

- High-performing products → Majority of advertising budget
- Medium-performing products → Seasonal campaigns
- Low-performing products → Limited promotions

This improves Marketing ROI.

---

# 10. Product Portfolio Optimization

Products can be divided into four groups.

### High Revenue + High Demand

Keep investing.

Examples:

- Redmi Smartphones
- Samsung Smartphones
- Mi Smart TVs

---

### High Price + Low Revenue

Need investigation.

Possible actions:

- Reduce price
- Improve marketing
- Bundle offers
- Improve product listing

---

### Low Price + High Demand

Excellent traffic generators.

Keep these products highly available.

---

### Low Price + Low Revenue

Consider:

- Removing them
- Replacing them
- Combining them into bundles

---

# Overall Business Conclusion

The Pareto Analysis clearly shows that the company's revenue is heavily concentrated in a small group of products.

Key observations include:

- Only **168 products (11.49%) generate nearly 80% of total Revenue Potential.**
- Electronics dominates the highest-performing products.
- Customer demand is a much stronger driver of revenue than product price alone.
- Many premium products have high prices but weak customer demand.
- Inventory, marketing, and business resources should primarily focus on the top-performing products to maximize profitability.

---

# Strategic Recommendations

Prioritize inventory for Top 168 products.

Increase advertising budget for high-performing Electronics.

Improve visibility of underperforming premium products.

Collect more customer reviews to improve demand.

Reduce inventory investment in long-tail products.

Consider discontinuing products with consistently low Revenue Potential.

Focus future product launches on categories similar to the current top performers.

Overall, the Pareto Analysis demonstrates that concentrating business efforts on a small percentage of high-value products can significantly improve revenue, operational efficiency, and return on investment.

# 4. Quality & Demand Validation (Rating vs. Price)

## Underpriced Stars vs. Overpriced Duds:

In [203]:
# --------------------------------------------
# Underpriced Stars
# High Rating + High Demand + Low Price
# --------------------------------------------

underpriced_stars = (
    df[
        (df["rating"] >= 4.5) &
        (df["rating_count"] >= df["rating_count"].quantile(0.75)) &
        (df["discounted_price"] <= df["discounted_price"].quantile(0.25))
    ]
    .sort_values(
        ["rating", "rating_count"],
        ascending=False
    )
)

underpriced_stars[
    [
        "product_name",
        "category",
        "actual_price",
        "discounted_price",
        "rating",
        "rating_count",
        "Revenue_Potential"
    ]
].head(20)

,product_name,category,actual_price,discounted_price,rating,rating_count,Revenue_Potential
820,Redgear MP35 Speed-Type Gaming Mousepad (Black...,Computers&Accessories,550.0,299.0,4.6,33434,9996766.0
40,AmazonBasics USB 2.0 Cable - A-Male to B-Male ...,Computers&Accessories,695.0,209.0,4.5,107687,22506583.0
852,AmazonBasics USB 2.0 Cable - A-Male to B-Male ...,Computers&Accessories,695.0,209.0,4.5,107686,22506374.0
212,AmazonBasics USB 2.0 Extension Cable for Perso...,Computers&Accessories,800.0,299.0,4.5,74977,22418123.0
30,AmazonBasics USB 2.0 - A-Male to A-Female Exte...,Computers&Accessories,750.0,199.0,4.5,74976,14920224.0
802,AmazonBasics USB 2.0 - A-Male to A-Female Exte...,Computers&Accessories,750.0,199.0,4.5,74976,14920224.0
612,"Dell MS116 1000Dpi USB Wired Optical Mouse, Le...",Computers&Accessories,650.0,299.0,4.5,33176,9919624.0
476,ELV Aluminum Adjustable Mobile Phone Foldable ...,Electronics,1499.0,269.0,4.5,28978,7795082.0
580,Elv Aluminium Adjustable Mobile Phone Foldable...,Electronics,1499.0,314.0,4.5,28978,9099092.0
615,"Duracell Ultra Alkaline AA Battery, 8 Pcs",Electronics,315.0,266.0,4.5,28030,7455980.0


In [204]:
# --------------------------------------------
# Overpriced Duds
# Low Rating + High Price + Heavy Discount
# --------------------------------------------

overpriced_duds = (
    df[
        (df["rating"] <= 3.5) &
        (df["actual_price"] >= df["actual_price"].quantile(0.75)) &
        (df["discount_percentage"] >= 50)
    ]
    .sort_values(
        ["rating", "discount_percentage"]
    )
)

overpriced_duds[
    [
        "product_name",
        "category",
        "actual_price",
        "discounted_price",
        "discount_percentage",
        "rating",
        "rating_count",
        "Revenue_Potential"
    ]
].head(20)

,product_name,category,actual_price,discounted_price,discount_percentage,rating,rating_count,Revenue_Potential
492,PTron Newly Launched Force X10 Bluetooth Calli...,Electronics,5999.0,1399.0,77.0,3.3,4415,6176585.0
463,PTron Newly Launched Force X10 Bluetooth Calli...,Electronics,5999.0,1299.0,78.0,3.3,4415,5735085.0
506,PTron Newly Launched Force X10 Bluetooth Calli...,Electronics,5999.0,1299.0,78.0,3.3,4415,5735085.0
82,SKYWALL 81.28 cm (32 inches) HD Ready Smart LE...,Electronics,19125.0,7299.0,62.0,3.4,902,6583698.0
867,Boult Audio Airbass Propods X TWS Bluetooth Tr...,Electronics,5999.0,1099.0,82.0,3.5,12966,14249634.0


In [205]:
import plotly.express as px

fig = px.scatter(
    df,
    x="discounted_price",
    y="rating",
    size="rating_count",
    color="category",
    hover_data=[
        "product_name",
        "actual_price",
        "discount_percentage",
        "Revenue_Potential"
    ],
    title="Rating vs Selling Price (Bubble Size = Demand)"
)

fig.update_layout(
    width=1300,
    height=700,
    legend_title="Category"
)

fig.show()

# 4. Quality & Demand Validation (Rating vs. Price)

## Objective
This analysis evaluates whether products are priced appropriately based on customer satisfaction (ratings) and market demand (rating count).

Two important business opportunities are identified:

- **Underpriced Stars:** Highly rated and highly demanded products sold at relatively low prices.
- **Overpriced Duds:** Expensive products with poor ratings despite offering heavy discounts.

Understanding these groups helps businesses optimize pricing, improve profitability, and identify products requiring quality improvements.

---

# A. Underpriced Stars

### Definition
Products satisfying:

- Rating ≥ **4.5**
- Rating Count in **Top 25%** (High Demand)
- Discounted Price in **Bottom 25%** (Low Selling Price)

These products have already proven that customers love them while demand remains consistently high.

---

## Observations

The analysis identified several products that receive:

- Excellent customer ratings (4.5–4.6)
- Extremely high customer engagement
- Very affordable selling prices
- Strong revenue potential

Examples include:

- AmazonBasics USB Cables
- Redgear Gaming Mousepad
- Dell USB Wired Mouse
- Duracell Batteries
- ELV Mobile Phone Stand

Most belong to:

- Computers & Accessories
- Electronics

---

## Business Interpretation

These products demonstrate strong product-market fit.

Customers perceive these products as:

- High quality
- Reliable
- Affordable
- Worth recommending

The large rating counts further confirm that demand is not accidental but sustained across many purchases.

---

## Business Opportunity

These products may be **underpriced**.

Since customers are already purchasing them in large volumes and giving excellent ratings, the company could potentially increase prices slightly without significantly reducing demand.

Even a small price increase can generate substantial additional revenue due to their high sales volume.

Example:

If a product selling 100,000 units annually increases price by only ₹20,

Additional Revenue

= 100,000 × ₹20

= ₹20,00,000

without increasing marketing costs.

---

## Key Insight

High customer satisfaction combined with high demand indicates pricing power.

Rather than relying solely on low prices, these products may support modest price increases while maintaining customer loyalty.

---

# B. Overpriced Duds

### Definition

Products satisfying:

- Rating ≤ **3.5**
- Actual Price in Top 25%
- Discount ≥ 50%

These products remain expensive even after large discounts while receiving poor customer ratings.

---

## Observations

The analysis identified products such as:

- PTron Force X10 Bluetooth Earphones
- Skywall Smart LED TV
- Boult Audio Airbass Earbuds

Characteristics:

- High original price
- Very heavy discounts (60–82%)
- Low customer ratings (3.3–3.5)

Despite aggressive discounts, customer satisfaction remains poor.

---

## Business Interpretation

Heavy discounts are failing to solve the underlying problem.

Possible reasons include:

- Poor product quality
- Performance issues
- Customer expectations not being met
- Strong competition offering better alternatives
- Misleading pricing strategy

Customers appear dissatisfied regardless of discount level.

---

## Business Risk

Continuing to heavily discount poorly rated products may:

- Reduce profit margins
- Damage brand reputation
- Increase product returns
- Generate negative reviews
- Reduce repeat purchases

Instead of increasing sales, excessive discounting may simply reduce profitability.

---

## Recommendation

Rather than increasing discounts further, the company should:

- Investigate customer reviews
- Improve product quality
- Reposition the product
- Consider discontinuing consistently poor performers

Fixing quality issues is likely to have a greater impact than lowering prices further.

---

# C. Rating vs. Selling Price Bubble Chart

## Chart Description

- **X-axis:** Discounted Price
- **Y-axis:** Product Rating
- **Bubble Size:** Rating Count (Demand)
- **Color:** Product Category

This visualization shows the relationship between selling price, customer satisfaction, and market demand.

---

## Major Insights

### 1. Weak Relationship Between Price and Rating

Products exist across all price ranges with ratings clustered around **4.0–4.4**.

This indicates:

- Higher prices do not necessarily lead to higher customer satisfaction.
- Customers value product quality more than price alone.

---

### 2. Demand is Concentrated Among Affordable Products

The largest bubbles are mostly located in the lower price range.

This suggests that:

- Lower-priced products attract significantly more customer purchases.
- Affordable pricing combined with good quality drives higher demand.

---

### 3. Premium Products Have Lower Sales Volume

Products priced above ₹20,000 generally have:

- Much smaller bubbles
- Similar ratings compared to cheaper products

This implies premium products receive fewer purchases, even when customer satisfaction is good.

---

### 4. Ratings are Consistently High Across Categories

Most categories maintain ratings between **4.0 and 4.4**, indicating:

- Overall product quality is relatively strong.
- Very few products receive extremely poor ratings.

---

### 5. Electronics and Computers Dominate High Demand

The largest bubbles mainly belong to:

- Electronics
- Computers & Accessories

These categories contribute the majority of customer engagement and revenue potential.

---

### 6. Few Extremely Expensive Products Exist

Several products priced between **₹40,000–₹80,000** appear as isolated points.

Although they maintain acceptable ratings, their small bubble sizes indicate:

- Limited customer demand
- Niche purchasing behavior
- Lower sales frequency

---

# Overall Business Conclusions

## Positive Findings

- Several products deliver excellent quality at affordable prices.
- Electronics and Computer Accessories dominate customer demand.
- Most products maintain customer ratings above 4.0.
- Affordable products generate the highest sales volume.

---

## Areas for Improvement

- Premium pricing alone does not guarantee customer satisfaction.
- Heavy discounts cannot compensate for poor product quality.
- Certain expensive products require quality improvements rather than additional price reductions.
- Underpriced high-performing products present opportunities for strategic price optimization.

---

# Final Recommendation

The company should adopt a differentiated pricing strategy:

- Slightly increase prices for high-demand, highly rated products to improve profit margins.
- Focus on improving quality instead of increasing discounts for poorly rated products.
- Continue investing in affordable, high-performing categories such as Electronics and Computers & Accessories.
- Monitor premium products closely to ensure higher prices are supported by superior customer experience and perceived value.

This approach balances profitability, customer satisfaction, and long-term revenue growth.

## Customer Friction Points:

In [206]:
# ----------------------------------------------------
# Customer Friction Points
# High Revenue Potential + Low Rating Count
# ----------------------------------------------------

customer_friction = (
    df[
        (df["discounted_price"] >= df["discounted_price"].quantile(0.75)) &
        (df["rating_count"] <= df["rating_count"].quantile(0.25))
    ]
    .sort_values(
        "discounted_price",
        ascending=False
    )
)

customer_friction.head(20)


,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count,Revenue_Potential,absolute_discounted_amount,discount_tier,Total_Discount_Leakage,Discount_Efficiency,Price_Realization_Ratio,Price_Realization_%
1014,B0B2RBP83P,"Lenovo IdeaPad 3 11th Gen Intel Core i3 15.6"" ...",Computers&Accessories,37247.0,59890.0,38.0,4.0,323,12030781.0,22643.0,30-50%,7313689.0,1.64,0.621924,62.192353
232,B09NNGHG22,Sansui 140cm (55 inches) 4K Ultra HD Certified...,Electronics,32990.0,56790.0,42.0,4.3,567,18705330.0,23800.0,30-50%,13494600.0,1.39,0.580912,58.091213
150,B0BB3CBFBM,VU 138 cm (55 inches) Premium Series 4K Ultra ...,Electronics,29990.0,65000.0,54.0,4.1,211,6327890.0,35010.0,50%+,7387110.0,0.86,0.461385,46.138462
286,B0B7B9V9QP,VU 108 cm (43 inches) Premium Series Full HD S...,Electronics,18999.0,35000.0,46.0,4.0,1001,19017999.0,16001.0,30-50%,16017001.0,1.19,0.542829,54.282857
112,B09YL9SN9B,LG 80 cm (32 inches) HD Ready Smart LED TV 32L...,Electronics,15990.0,23990.0,33.0,4.3,1035,16549650.0,8000.0,30-50%,8280000.0,2.00,0.666528,66.652772
299,B0BNDD9TN6,WANBO X1 Pro (Upgraded) | Native 1080P Full HD...,Electronics,13990.0,28900.0,52.0,4.5,7,97930.0,14910.0,50%+,104370.0,0.94,0.484083,48.408304
1208,B08MZNT7GP,Havells OFR 13 Wave Fin with PTC Fan Heater 29...,Home&Kitchen,12499.0,19825.0,37.0,4.1,322,4024678.0,7326.0,30-50%,2358972.0,1.71,0.630467,63.046658
206,B09ZPJT8B2,TCL 80 cm (32 inches) HD Ready Certified Andro...,Electronics,11990.0,31990.0,63.0,4.2,64,767360.0,20000.0,50%+,1280000.0,0.60,0.374805,37.480463
308,B09NNJ9WYM,Sansui 80cm (32 inches) HD Ready Smart LED TV ...,Electronics,10990.0,19990.0,45.0,3.7,129,1417710.0,9000.0,30-50%,1161000.0,1.22,0.549775,54.977489
77,B09ZPM4C2C,TCL 80 cm (32 inches) HD Ready Certified Andro...,Electronics,10901.0,30990.0,65.0,4.1,398,4338598.0,20089.0,50%+,7995422.0,0.54,0.351759,35.175863


In [207]:
import plotly.express as px

fig = px.scatter(
    customer_friction,
    x="rating_count",
    y="Revenue_Potential",
    size="discounted_price",
    color="category",
    hover_data=[
        "product_name",
        "rating",
        "discounted_price",
        "actual_price"
    ],
    title="Customer Friction Points (High Revenue Potential, Low Reviews)"
)

fig.update_layout(
    width=1300,
    height=700
)

fig.show()


## Objective

The objective of this analysis is to identify **high-value products that have strong business potential but lack sufficient customer reviews**. Products with few ratings often struggle to gain customer trust, reducing their visibility and conversion rate despite offering good value.

Unlike revenue analysis, this approach focuses on products that deserve greater exposure through marketing and review-generation strategies.

---

## Business Logic

Customer friction occurs when a product has:

- High selling price (high-value product)
- Good customer rating
- Low number of customer reviews

These products may be hidden opportunities because customers hesitate to purchase items that lack sufficient social proof.

---

## Selection Criteria

A product is classified as a **Customer Friction Point** if it satisfies:

- Rating ≥ **4.2**
- Rating Count ≤ **25th Percentile**
- Discounted Price ≥ **Median Selling Price**

This helps identify premium products that customers appreciate but have not yet received enough visibility.

---

## Visualization

**Scatter Plot**

- **X-axis:** Rating Count
- **Y-axis:** Discounted Price
- **Bubble Size:** Revenue Potential
- **Color:** Product Category

This visualization highlights expensive products with relatively few customer reviews.

---

## Key Insights

### 1. Premium Products with Low Visibility

Several expensive products have received very few customer reviews despite maintaining strong customer ratings.

**Business Impact**

- Low review count reduces customer confidence.
- Products receive lower organic visibility on e-commerce platforms.
- Sales potential remains underutilized.

---

### 2. Lack of Social Proof

Customers often compare review counts before purchasing.

Products with:

- High ratings
- Premium pricing
- Very few reviews

are more likely to experience lower conversion rates than similar products with thousands of reviews.

---

### 3. Marketing Opportunity

These products represent excellent candidates for promotional campaigns.

Possible initiatives include:

- Sponsored advertisements
- Homepage recommendations
- Email campaigns
- Influencer marketing
- Featured product sections

Increasing visibility can significantly improve review generation and sales.

---

### 4. Review Generation Opportunity

Since these products already maintain good ratings, businesses should encourage verified buyers to leave reviews.

Possible strategies include:

- Review reminder emails
- Cashback for verified reviews
- Loyalty reward points
- Post-purchase notifications

More reviews increase buyer confidence and improve product ranking.

---

### 5. Category-Level Analysis

Certain categories may contain more customer friction products than others.

This indicates that some categories suffer from visibility problems rather than product quality problems.

Marketing investment can therefore be allocated more efficiently.

---

## Business Recommendations

- Promote high-rated products with low review counts through targeted campaigns.
- Encourage verified customers to submit product reviews.
- Improve product visibility using recommendation engines.
- Highlight these products during seasonal sales.
- Monitor review growth after promotional campaigns.
- Prioritize customer friction products in digital marketing strategies.

---

## Business Value

Identifying customer friction points enables businesses to:

- Increase product visibility
- Improve customer trust
- Generate additional reviews
- Increase conversion rates
- Improve organic product rankings
- Unlock hidden revenue opportunities without changing product pricing

---

## Conclusion

This analysis identifies **high-quality products that are underperforming due to insufficient customer reviews rather than poor product quality**.

Instead of investing in product improvements, businesses should focus on increasing product exposure and generating more verified customer reviews. Strengthening social proof for these products can significantly improve conversions, customer trust, and overall revenue.